tests/models/gptsan_japanese/test_modeling_gptsan_japanese.py

```
@@ -0,0 +1,327 @@
+                -9.814510345458984,
+                -9.996501922607422,
+                -11.375690460205078,
+                -10.113405227661133,
+                -10.546867370605469,
+                -10.04369068145752,
+                -10.907809257507324,
+                -10.504216194152832,
+                -11.129199028015137,
+                -10.151124000549316,
+                -21.96586799621582,
+                -9.086349487304688,
+                -11.730339050292969,
+                -10.460667610168457,
+                -10.298049926757812,
+                -10.784148216247559,
+                -10.840693473815918,
+                -22.03152847290039,
+            ],
+        ]
+        target = np.array(target).flatten()
+        predict = output_logits[0, :, :20].flatten()
+
+        def check(a, b, epsilon=5e-4):
+            return abs(a - b) < epsilon * max(abs(a), abs(b))
+
+        assert np.all([check(target[i], predict[i]) for i in range(len(target))])
+
+    @tooslow
+    def test_sample(self):
+        model = GPTSANJapaneseForConditionalGeneration.from_pretrained("Tanrei/GPTSAN-japanese")
+        tokenizer = GPTSANJapaneseTokenizer.from_pretrained("Tanrei/GPTSAN-japanese")
+        # Output of original model created with mesh-tensoflow
+        target = [
+            ("武田信玄は", 35675),
+            ("武田信玄は、", 45),
+            ("武田信玄は、この", 29),
+            ("武田信玄は、このよう", 30642),
+            ("武田信玄は、このような", 35680),
+            ("武田信玄は、このような「", 8640),
+            ("武田信玄は、このような「武田", 31617),
+            ("武田信玄は、このような「武田家", 30646),
+            ("武田信玄は、このような「武田家の", 31617),
+            ("武田信玄は、このような「武田家の家", 31381),
+        ]
+        for input, output in target:
+            input_ids = tokenizer.encode(input, return_tensors="pt")
+            outputs = model(input_ids)
+            output_logits = outputs.logits.detach().cpu().numpy()[0]
+            output_id = np.argmax(output_logits[-1])
+            assert output_id == output
```

Let's rather raise an error here